src/SelfServe/SelfServeUtils.tsx

```diff
@@ -165,6 +187,9 @@ const getInput = (value: CommonInputTypes): AnyInput => {
       }
       return value as NumberInput;
     case "string":
+      if (value.description) {
```

Comment:
Consider having a separate type for description, so we avoid having to two code paths everywhere. Also note that it seems that in the other file, we test slightly differently (`if ("description" in input)`)

Reply:
value.description here is fine since value belongs to the CommonInputType
the check ("description" in input) is to make sure the casting is fine.
Also this is to make sure description is only set to a parameter in self serve class that is a string.